cluster-autoscaler/processors/nodegroupset/aws_nodegroups_test.go

```diff
@@ -0,0 +1,181 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodegroupset
+
+import (
+	"testing"
+
+	"k8s.io/autoscaler/cluster-autoscaler/context"
+	. "k8s.io/autoscaler/cluster-autoscaler/utils/test"
+)
+
+func TestIsAwsNodeInfoSimilar(t *testing.T) {
+	comparator := CreateAwsNodeInfoComparator([]string{})
+	node1 := BuildTestNode("node1", 1000, 2000)
+	node2 := BuildTestNode("node2", 1000, 2000)
+
+	for _, tc := range []struct {
+		description    string
+		label          string
+		value1         string
+		value2         string
+		removeOneLabel bool
```

Comment:
nit: I would consider adding negative cases as well. Right now a comparator that strips all labels (or just always return true) would pass the tests. 

I don't think it's at all blocking for this PR (it's already very much net positive on test coverage), just something to maybe consider later on.